core/metacling/src/TCling.cxx

```diff
@@ -1437,8 +1437,14 @@ TCling::TCling(const char *name, const char *title, const char* const argv[])
    fClingCallbacks->SetAutoParsingSuspended(fIsAutoParsingSuspended);
    fInterpreter->setCallbacks(std::move(clingCallbacks));
 
-   // We are set up.
-   EnableAutoLoading();
+   if (!fromRootCling) {
+      // Make sure cling looks into ROOT's libdir, even if not part of LD_LIBRARY_PATH
+      // e.g. because of an RPATH build.
+      fInterpreter->getDynamicLibraryManager()->addSearchPath(TROOT::GetLibDir().Data());
+
+      // We are set up.
+      EnableAutoLoading();
```

Comment:
We have an early exit if we are in rootcling mode: https://github.com/root-project/root/blob/master/core/metacling/src/TCling.cxx#L2931

Reply:
Thanks, well spotted! As I favor readability over having to remember or looking up I'll just leave it if you don't mind - it shouldn't be expensive.

Reply:
Wouldn't a comment be as readable as adding 'redundant' calls?

Reply:
If both of you insist... I do find a comment less visible than code, but I did what the two of you suggested.